### PR TITLE
Fix F# indexers resolution in C#

### PR DESCRIPTION
**Pull request by DedSec256**
https://youtrack.jetbrains.com/issue/RIDER-42954

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/MemberDeclaration.cs [hunk 1]
```diff
@@ -66,7 +65,9 @@ private IDeclaredElement CreateProperty(FSharpMemberOrFunctionOrValue mfv)
     {
       foreach (var accessor in AccessorDeclarationsEnumerable)
         if (accessor.IsExplicit)
-          return new FSharpPropertyWithExplicitAccessors(this);
+          return SourceName == StandardMemberNames.DefaultIndexerName && !IsStatic
```
Comment (DedSec256) [inline ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/MemberDeclaration.cs:68]: we can add the IsIndexer flag to the IMemberDeclaration

**Review by auduchinok — COMMENTED** [review]
`Equals` override is a bit questionable, otherwise looks good. Thanks for adding the parser tests too!

File: ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/MemberDeclaration.cs (hunk 1 above)
Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/MemberDeclaration.cs:68]: Yup, sounds reasonable, but it's OK if done later.

I'd probably also split member declarations to be different types for methods and properties.

**Review by auduchinok — APPROVED** [review]
Thanks! Looks good to me now.

**Pull request merged**